hosts/galileo/__init__.py

```diff
@@ -0,0 +1,395 @@
+
+### TODO: finish these routines...
+
+    @staticmethod
+    def spice_instrument_kernel(inst, asof=None):
+        """A dictionary containing the Instrument Kernel information.
+
+        Also furnishes it for use by the SPICE tools.
+
+        Input:
+            inst        one of "ISS", "UVIS", "VIMS", "CIRS", etc.
+            asof        an optional date in the past, in ISO date or date-time
+                        format. If provided, then the information provided will
+                        be applicable as of that date. Otherwise, the most
+                        recent information is always provided.
+
+        Return:         a tuple containing:
+                            the dictionary generated by textkernel.from_file()
+                            the name of the kernel.
+        """
+        if asof is not None:
+            (day,sec) = julian.day_sec_from_iso(stop_time)
+            asof = julian.ymdhms_format_from_day_sec(day, sec)
+
+        spicedb.open_db()
+        kernel_info = spicedb.select_inst(-77, types='IK', inst=inst, asof=asof)
+        spicedb.furnish_kernels(kernel_info, fast=True)
+        spicedb.close_db()
+
+        return (spicedb.as_dict(kernel_info), spicedb.as_names(kernel_info)[0])
+
+    #===========================================================================
+    @staticmethod
+    def spice_frames_kernel(asof=None):
+        """A dictionary containing the Galileo Frames Kernel information.
+
+        Also furnishes the kernels for use by the SPICE tools.
+
+        Input:
+            asof        an optional date in the past, in ISO date or date-time
+                        format. If provided, then the information provided will
+                        be applicable as of that date. Otherwise, the most
+                        recent information is always provided.
+
+        Return:         a tuple containing:
+                            the dictionary generated by textkernel.from_file()
+                            an ordered list of the names of the kernels
+        """
+        if asof is not None:
+            (day,sec) = julian.day_sec_from_iso(stop_time)
+            asof = julian.ymdhms_format_from_day_sec(day, sec)
+
+        spicedb.open_db()
+        kernel_list = spicedb.select_inst(-77, types='FK', asof=asof)
+        spicedb.furnish_kernels(kernel_info, fast=True)
+        spicedb.close_db()
+
+        return (spicedb.as_dict(kernel_list), spicedb.as_names(kernel_list))
+
+    #===========================================================================
+    @staticmethod
+    def used_kernels(time, inst, return_all_planets=False):
+        """The list of kernels associated with a Galileo observation at a
+        selected range of times.
+        """
+        if return_all_planets:
+            bodies = [1, 199, 2, 299, 3, 399, 4, 499, 5, 599, 6, 699,
+                      7, 799, 8, 899]
+            if time[0] >= TOUR:
+                bodies += Body.JUPITER_MOONS_LOADED
```

Comment:
This code doesn't make any sense

Reply:
Agreed, this is the result of a search and replace.  This was one of the TODO methods at the end of the file that I have deleted in response to an earlier comment.  As far as I can tell, although the output of this method is added to the observation, it is never used for anything (even in Cassini ISS), and it's not clear what it's purpose is, because it doesn't actually reflect which kernels are used.

Indeed, I don't currently use this in Juno, so deleteing it here is in the spirit of  'on par with Junocam'.

Reply:
I agree this is a useful thing to have, but the list returned by this function has no connection to which kernels were actually used to create the observation.  What is really needed is for whatever function that loads the kernel to record their names at that time.  This could be done now, but will likely be part of the larger kernel management project.

Reply:
I don't know why you say that. It works great for Cassini ISS. Why wouldn't it work for Galileo? Yes, it includes all of the moon kernels, but the important part is which pointing kernels were used.

This code returns the kernels needed for the time range of the observation:

```
        return spicedb.used_basenames(time=time, inst=inst, sc=-77,
                                      bodies=bodies)
```

Reply:
Restored and updated used_kernels() for galileo.